### PR TITLE
Interactive expansion of sidebar section contents

### PR DESCRIPTION
**Pull request by scopegate**
This allows the sections in the sidebar to be expanded in place, so users can browse the contents of a section without having to click into it. 

To-do:
- [x] Add some animation
- [x] Check and improve handling of the agent version logic (ie /v2 vs /v3)
- [ ] Add icons against sections?

### CODE REVIEW

File: app/views/layouts/_sidebar.html.erb [hunk 1]
```diff
@@ -0,0 +1,272 @@
+<nav class="Docs__nav">
+  <div class="Docs__nav__section-container" data-docs-path="tutorials">
+    <p class="Docs__nav__section-heading">Tutorials <%= image_tag('nav-exp-icon.svg') %></p>
+    <div class="Docs__nav__section-content">
+      <ul class="Docs__nav__sub-nav">
+        <li class="Docs__nav__sub-nav__item"><%= sidebar_link_to "Getting Started", 'tutorials/getting-started' %></li>
```
Comment (blaknite) [inline app/views/layouts/_sidebar.html.erb:6]: I reckon we should turn the rest of the sidebar structure into helpers. There's a lot of repetition going on here. Maybe `sidebar_section` and `sidebar_item`? Personally I'd merge the `ul` into the `sidebar_section` helper and put the `li` into the existing `sidebar_link_to` and rename it.

The `sidebar_section` helper is going to need to take a block. How's your grasp of blocks? Let me know if you want to pair.

**Comment by blaknite** [comment]
Regarding the JS, it does what it says on the tin. Maybe it could be tidier but I'm plenty happy to ship that as is. @ticky may have more to add though.

File: app/views/layouts/_sidebar.html.erb (hunk 1 above)
Comment (scopegate) [reply app/views/layouts/_sidebar.html.erb:6]: Great suggestion! It definitely felt a little off to be manually shunting around this much markup. Given we don't kick off YAML proper until Tuesday, maybe we could pair on this sometime Monday?

Comment (scopegate) [reply app/views/layouts/_sidebar.html.erb]: I've wrapped the `li` into the `sidebar_link_to`, but looking at the other sections I'm not sure there's much boilerplate to remove in creating a section helper, as the sections are actually quite different - some have one list within them, some have multiple lists with subheadings, and the Agent section actually has two completely separate sets of content. 🤔

**Comment by plaindocs** [comment]
@scopegate what state is this in? Can we do a minimal amount of work and push it out?

**Pull request closed without merging**